### PR TITLE
Label support for networks on creation

### PR DESCRIPTION
**Pull request by ehazlett**
This adds label support for networks on creation.

Refs: https://github.com/docker/docker/pull/21270

### CODE REVIEW

File: network.go [hunk 1]
```diff
@@ -1267,3 +1294,10 @@ func (n *network) IPv6Enabled() bool {
 
 	return n.enableIPv6
 }
+
+func (n *network) Labels() map[string]string {
+	n.Lock()
+	defer n.Unlock()
+
+	return n.labels
```
Comment (chenchun) [inline network.go]: I think it's better to make a copy of labels map.

**Comment by mavenugo** [comment]
@ehazlett i agree with @chenchun. 

LGTM otherwise.

**Comment by mavenugo** [comment]
LGTM

**Comment by aboch** [comment]
@ehazlett 
Can you please rebase to latest and push again so that you pick the fix for the circle-ci.
Given you are pushing again, do you mind squashing the two commits in one ?
Thanks

**Comment by ehazlett** [comment]
@aboch done.  thanks!

**Comment by aboch** [comment]
LGTM

**Pull request merged**